src/versions.jl

```diff
@@ -273,6 +275,42 @@ end
 Base.show(io::IO, s::VersionSpec) = print(io, "VersionSpec(\"", s, "\")")
 
 
+"""
+    compressed_versionspec(pool, subset=pool)
+
+ - `pool`: all versions that exist (a superset of `subset`)
+ - `subset`: the subset of those versions that we want to include in the spec.
+
+Works out a compressed `VersionSpec` that permits everything in the `subset`,
+and nothing else from the `pool`.
+"""
+function compressed_versionspec(pool, subset=pool)
```

Comment:
Pkg-Dev call wants to rename this because it is not quite the same as what compress means in the registry tools.
Maybe: `ranged_versionspec` ?

Reply:
`versionspec_range_string`?